credsweeper/utils/util.py

```diff
@@ -198,6 +198,34 @@ def patch2files_diff(raw_patch: List[str], change_type: str) -> Dict[str, List[D
             logger.error(f"Change type should be one of: 'added', 'deleted'; but received {change_type}")
         return {}
 
+    @staticmethod
+    def preprocess_diff_rows(
+            added_line_number: Optional[int],  #
+            deleted_line_number: Optional[int],  #
+            line: str) -> List[DiffRowData]:
+        """Auxiliary function to extend diff changes.
+
+        Args:
+            added_line_number: number of added line or None
+            deleted_line_number: number of deleted line or None
+            line: the text line
+
+        Return:
+            diff rows data with as list of row change type, line number, row content
+
+        """
+        rows_data = []
+        if deleted_line_number is None:
+            # indicates line was inserted
+            rows_data.append(DiffRowData(DiffRowType.ADDED, added_line_number, line))
+        elif added_line_number is None:
+            # indicates line was removed
+            rows_data.append(DiffRowData(DiffRowType.DELETED, deleted_line_number, line))
+        else:
+            rows_data.append(DiffRowData(DiffRowType.ADDED_ACCOMPANY, added_line_number, line))
+            rows_data.append(DiffRowData(DiffRowType.DELETED_ACCOMPANY, deleted_line_number, line))
```

Comment:
I've looked at this code and I don't know why it needed.

As I know CredSweeper only use `DiffRowType.ADDED` and `DiffRowType.DELETED` on [`__main__.py` L#285, L#290](https://github.com/Samsung/CredSweeper/blob/main/credsweeper/__main__.py#L285-L290).

Please check this.

Reply:
Yes, it is. 
Only "added" and "deleted" are used but in the case we can obtain a case when added and deleted line number are present. The author has added *_ACCOMPANY types. Maybe for debug. It can be  reduced.

I've fuzzed the patch provider (actually, whatthepatch library) and found many issues which lead to uncaught exceptions (if Credsweeper is used as class).

Additionally, DiffRowType was changed to Enum.

Reply:
Okay I've checked all changes, and it sounds good to found many issues from `whatthepatch` :)